hera_qm/xrfi.py

```diff
@@ -679,6 +679,38 @@ def watershed_flag(uvf_m, uvf_f, nsig_p=2., nsig_f=None, nsig_t=None, avg_method
     return uvf
 
 
+def _ws_flag_1D(metric, fin, nsig=2.):
```

Comment:
I actually just noticed the `_ws_flag_waterfall` function does take 1D arrays. Perhaps we should replace the code there with your code instead of making a redundant function?

Reply:
On second thought... your code does something slightly different -- it does not iterate until the region has stopped growing. Was this intentional? And should we rename some functions to make it more clear?
Maybe we still put your code in the original function, but add an option to not iterate?

Reply:
I missed the while statement in the original ws_flagging. I've harmonized the two into a single, more elegant function.